### ISSUE TITLE
[no-type-alias] <issue title> incompatible with `readonly`

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#813]
**Repro**

```JSON
{
    "rules": {
        "@typescript-eslint/no-type-alias": [
            2,
            {
                "allowAliases": "in-unions-and-intersections",
                "allowCallbacks": "always",
                "allowLiterals": "in-unions-and-intersections",
                "allowMappedTypes": "always"
            }
        ]
    }
}
```

```TS
// your repro code case
export type Arrayable<T> = T[] | readonly T[]
```

**Expected Result**
No report

**Actual Result**
```
✘  @typescript-eslint/no-type-alias

   Unhandled in union types are not allowed
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0-alpha.4` |
| `@typescript-eslint/parser`        | `2.0.0-alpha.4` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `v12.7.0` |
| `npm`                              | `6.10.2` |

**Comment by ankeetmaini** [comment]
This is being worked on in #775
